modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreator.java

```diff
@@ -116,7 +124,12 @@ public Subject createSubject(SecurityDomainContext ctx, String name, String pass
             }
          }
 
-      } catch (RealmUnavailableException e) {
+      } catch (RealmUnavailableException | InvalidKeyException | InvalidKeySpecException | NoSuchAlgorithmException e) {
+         if (e instanceof RealmUnavailableException) {
+            SECURITY_LOGGER.realmNotAvailable(principal.getName());
+         } else {
```

Comment:
```
} catch (RealmUnavailableException e) {
 SECURITY_LOGGER.realmNotAvailable(principal.getName());
}catch( InvalidKeyException | InvalidKeySpecException | NoSuchAlgorithmException ex) {
 SECURITY_LOGGER.plainTextPasswordMustBeRecoverable();
}
```
Is adding more catch block better ?  It doesn't need to add more instance checks.

Reply:
@jimma Updated now